app/components/SocketProvider.js

```diff
@@ -173,6 +175,15 @@ class SocketProvider extends React.Component<Props> {
 
           if (event.event === "collections.delete") {
             documents.removeCollectionDocuments(collectionId);
+            memberships.removeCollectionMemberships(collectionId);
+
+            const collection = collections.get(collectionId);
+            if (collection) {
+              collection.deletedAt = collectionDescriptor.updatedAt;
+            }
+
+            collections.remove(collectionId);
+            policies.remove(collectionId);
```

Comment:
ended up handling this case the same way we do for documents, and letting the component decide what to do with a collection where `deletedAt` is present